rules/always-return.js

```diff
@@ -124,6 +124,34 @@ function peek(arr) {
   return arr[arr.length - 1]
 }
 
+/**
+ * Checks if the node is an assignment to an ignored variable
+ * @param {Node} node
+ * @param {string[]} ignoredVars
+ * @returns {boolean}
+ */
+function isIgnoredAssignment(node, ignoredVars) {
+  if (node.type !== 'ExpressionStatement') return false
+  const expr = node.expression
+  if (expr.type !== 'AssignmentExpression') return false
+  const left = expr.left
+  // istanbul ignore else
+  if (left.type === 'MemberExpression') {
+    if (left.object.type === 'Identifier') {
+      return ignoredVars.includes(left.object.name)
+    }
+    if (
+      left.object.type === 'MemberExpression' &&
+      left.object.object.type === 'Identifier'
+    ) {
+      const fullName = `${left.object.object.name}.${left.object.property.name}`
+      return ignoredVars.includes(fullName)
+    }
+  }
```

Comment:
**question(non-blocking):** Could this be done recursively?

The `<root>.<properties>` examples in the docs makes it feel like we could add more properties, eg `<root>.<property-1>.<property-2>.<property-3>`

Reply:
An example function for this is, not 100% sure it'll work
```js
function literalMemberExpressionText(input) {
  if (input.type === "Identifier") {
    return [input.name];
  }

  if (
    input.computed === true &&
    input.property.type !== "Literal"
  ) {
    return null;
  }

  const parents = literalMember(input.object);
  if (parents === null) {
    return null;
  }

  return [
    ...literalMember(input.object),
    input.property.name || input.property.value
  ];
}
```

Reply:
I'll take a look at this, I don't have an opinion but it seems natural that if we add `"window"` window.x.y is also not blocked, which isn't the case right now